root/etc/e-smith/templates/etc/ssh/sshd_config/70Restricted2Sftp

```diff
@@ -3,16 +3,31 @@
     my $domain = $DomainName;
     my @SftpRestricted = ();
 
+    my $admin = $admins{'group'} || 'domain admins';
+    $admin .= "\@$DomainName" if ($admin !~ '@');
+
     foreach ( split(',',$sshd{'AllowGroups'} || '')) {
         my ($group, $sftp) = split(':', $_);
-        if(!$group || !$sftp) {
+        if(!$group) {
             next;
         }
+        if (!$sftp) {
+            $sftp = '';
+        }
+
         $group .= "\@$domain" if ($group !~ '@');
-        push @SftpRestricted, $group if ($sftp eq 'sftp');
+        if ($sftp eq 'sftp') {
+            push @SftpRestricted, $group;
+        } else {
+            # the group won't be forced to use sftp
+            # in case of membership of several groups
+            push @SftpRestricted, '!'.$group;
+       }
     }
 
     if (@SftpRestricted) {
+        #domain admins cannot be forced to SFTP
+        push @SftpRestricted, '!'.$admin;
```

Comment:
We can remove these lines, if the suggestion for line 6-7 is accepted. The idea is to code for `$admins_group` like any other group.

```suggestion
```

Reply:
nice